crates/proto-flow/src/flow.rs

```diff
@@ -896,6 +896,12 @@ pub mod combine_api {
         /// TODO(johnny): This is a kludge as we seek to remove this API.
         #[prost(message, repeated, tag = "5")]
         pub projections: ::prost::alloc::vec::Vec<super::Projection>,
+        /// The name of the collection that's being written to.
+        #[prost(string, tag = "6")]
+        pub collection_name: ::prost::alloc::string::String,
+        /// Whether to emit maximally-constrained schemas as documents are combined.
```

Comment:
🤔 what about `infer_schema_json: string`, which is a json-encoded string of JSON to start inferring from, or empty if inference shouldn't happen?

this is a generalization of the boolean which allows the caller to send `false` as the starting schema, but _also_ allows for some other schema as the starting point.

Reply:
background: i think we'll be plumbing the inferred schema through into the collection spec, which means we'd be able to start from the schema that was inferred as-of the last publication of the collection, which would make the inferred schema changes more meaningful in the task logs.

Reply:
Done 👍